nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java

```diff
@@ -59,96 +60,76 @@
 
     protected static final String STATE = ":state";
 
-    protected static final String SCROLLED_DOCUMENT_COUNT = ":scrolledDocumentCount";
-
-    protected final BulkServiceDescriptor descriptor;
-
-    /** Initialized in a lazy way. */
-    protected KeyValueStore kvStore;
-
-    /** Initialized in a lazy way. */
-    protected LogManager logManager;
-
-    public BulkServiceImpl(BulkServiceDescriptor descriptor) {
-        this.descriptor = descriptor;
-    }
+    protected static final String SCROLLED_DOCUMENT_COUNT = ":count";
 
     @Override
-    public BulkStatus runOperation(BulkCommand command) {
+    public BulkStatus runAction(BulkCommand command) {
         if (log.isDebugEnabled()) {
-            log.debug("Run operation with command=" + command);
+            log.debug("Run action with command=" + command);
         }
-        // fill command object
+        // check command
         if (isEmpty(command.getRepository())) {
-            String repository = Framework.getService(RepositoryManager.class).getDefaultRepositoryName();
-            command.withRepository(repository);
+            throw new IllegalArgumentException("Missing mandatory values");
         }
-        // create the operation id and status
-        UUID bulkOperationId = UUID.randomUUID();
+        // create the action id and status
+        String bulkActionId = UUID.randomUUID().toString();
 
         BulkStatus status = new BulkStatus();
-        status.setUUID(bulkOperationId);
+        status.setId(bulkActionId);
         status.setState(SCHEDULED);
-        status.setCreationDate(ZonedDateTime.now());
+        status.setCreationInstant(Instant.now());
         status.setCommand(command);
 
         try {
-            byte[] commandAsBytes = OBJECT_MAPPER.writeValueAsBytes(command);
+            byte[] commandAsBytes = MarshallerHelper.objectToJson(command, RenderingContext.CtxBuilder.get())
+                                                    .getBytes();
 
             // store the bulk command and status in the key/value store
             KeyValueStore keyValueStore = getKvStore();
-            keyValueStore.put(bulkOperationId + STATE, status.getState().toString());
-            keyValueStore.put(bulkOperationId + CREATION_DATE, status.getCreationDate().toString());
-            keyValueStore.put(bulkOperationId + COMMAND, commandAsBytes);
+            keyValueStore.put(bulkActionId + STATE, status.getState().toString());
+            keyValueStore.put(bulkActionId + CREATION_DATE, status.getCreationInstant().toString());
+            keyValueStore.put(bulkActionId + COMMAND, commandAsBytes);
```

Comment:
I though we only need to store the state in the kv store not the command itself

Reply:
We store it in order to get it for getting the status. Also, we send this command in each record in the system. It's not really a good thing right ? Should we get it directly from K/V ? Don't know if it's better than sending it in record.

Reply:
Choice made is to keep command in K/V, computation will ask for it at beginning of processing.